### PR TITLE
snap-bootstrap: remove create-partitions and update tests

### PR DESCRIPTION
**Pull request by mvo5**
This commit removes the `snap-bootstrap create-partitions` command
and move it into the tests so that we can still integration test
the partition creation.

Based on https://github.com/snapcore/snapd/pull/8547

### CODE REVIEW

File: tests/main/uc20-create-partitions-reinstall/task.yaml [hunk 1]
```diff
@@ -1,8 +1,14 @@
...
+summary: Integration tests for the bootstrap.Run autodetect
 
 # use the same system and tooling as uc20
 systems: [ubuntu-20.04-64]
 
+environment:
+    # an empty $topsrcdir/tests/go.mod seems to break importing or building go
+    # packages referenced by their import paths while under the tests directory,
+    # need to disable go modules support for this test
+    GO111MODULE: off
```
Comment (pedronis) [inline tests/main/uc20-create-partitions-reinstall/task.yaml:10]: this is both setting this and removing go.mod, I suppose go.mod should be reinstated?

Comment (mvo5) [reply tests/main/uc20-create-partitions-reinstall/task.yaml:10]: Sorry for that, not sure how this happend

File: tests/main/uc20-create-partitions-reinstall/task.yaml [hunk 2]
```diff
@@ -45,7 +51,8 @@ execute: |
     udevadm info --query=property "${LOOP}p2" | grep ID_FS_TYPE ||:
 
     echo "Run the snap-bootstrap tool in auto-detect mode"
-    /usr/lib/snapd/snap-bootstrap create-partitions ./gadget-dir
+    go get ../../lib/create-partitions
```
Comment (bboozzoo) [inline tests/main/uc20-create-partitions-reinstall/task.yaml:54]: Should the binary be named `uc20-create-partitions` instead?

Comment (bboozzoo) [inline tests/main/uc20-create-partitions-reinstall/task.yaml:54]: Perhaps we should tread it like `gendeveloper1model` and `go get` the binary in prepare.sh.

File: tests/nested/manual/uc20-create-partitions-encrypt/task.yaml [hunk 3]
```diff
@@ -52,10 +58,13 @@ prepare: |
 
 execute: |
     LOOP="$(cat loop.txt)"
+
     echo "Run the snap-bootstrap tool"
-    /usr/lib/snapd/snap-bootstrap create-partitions --mount \
+    go get ../../../lib/create-partitions
+    create-partitions --mount \
```
Comment (cmatsuoka) [inline tests/nested/manual/uc20-create-partitions-encrypt/task.yaml:64]: We could also test the creation of the TPM lockout autorization file and the policy update data file.

File: tests/main/uc20-create-partitions-reinstall/task.yaml [hunk 4]
```diff
@@ -1,8 +1,14 @@
-summary: Integration tests for the snap-bootstrap binary autodetect
+summary: Integration tests for the bootstrap.Run autodetect
 
 # use the same system and tooling as uc20
 systems: [ubuntu-20.04-64]
```
Comment (cmatsuoka) [inline tests/main/uc20-create-partitions-reinstall/task.yaml:4]: Key sealing is always performed when encrypting and it requires secure boot and tpm to be enabled, so we can use `ubuntu-20.04-64-secboot` here (after snapcore/spread#104 lands and replacing/reconciling with #8539).

**Comment by cmatsuoka** [comment]
Update the encrypted partition test with changes from #8539, which requires spread with snapcore/spread#104.

**Comment by mvo5** [comment]
Hm, this now fails in:
```
2020-05-06T08:21:14.6859780Z ++ get_image_url_for_nested_vm
2020-05-06T08:21:14.6859871Z ++ [[ google == google* ]]
2020-05-06T08:21:14.6859979Z ++ get_google_image_url_for_nested_vm
2020-05-06T08:21:14.6860083Z ++ case "$SPREAD_SYSTEM" in
2020-05-06T08:21:14.6860342Z ++ echo 'unsupported system'
2020-05-06T08:21:14.6860439Z ++ exit 1
2020-05-06T08:21:14.6860710Z + wget 'unsupported system' -O cloudimg.img
2020-05-06T08:21:14.6861062Z --2020-05-06 08:21:14--  http://unsupported%20system/
2020-05-06T08:21:14.6861215Z Resolving unsupported system (unsupported system)... failed: Name or service not known.
2020-05-06T08:21:14.6868541Z wget: unable to resolve host address â€˜unsupported systemâ€™
2020-05-06T08:21:14.6869065Z -----
2020-05-06T08:21:14.6869181Z .
2020-05-06T08:21:15.3010007Z 2020-05-06 08:21:15 Debug output for google:ubuntu-20.04-64-secboot:tests/main/preseed (may060748-687839)
```

**Comment by cmatsuoka** [comment]
Fixed as described by @sergiocazzolato. Note that tests won't pass until spread is prepared to provision systems with secure boot and tpm enabled.

**Pull request merged**